src/nvim/message.c

```diff
@@ -1034,16 +1133,62 @@ static void msg_hist_add_multihl(HlMessage msg, bool temp)
 
   if (msg_hist_off || msg_silent != 0) {
     hl_msg_free(msg);
-    return;
+    return -1;
   }
 
-  // Allocate an entry and add the message at the end of the history.
-  MessageHistoryEntry *entry = xmalloc(sizeof(MessageHistoryEntry));
-  entry->msg = msg;
+  bool is_kind_progress = msg_ext_kind != NULL && strcmp(msg_ext_kind, MSG_KIND_PROGRESS) == 0;
```

Comment:
use strequal() wherever appropriate, especially when you are checking "not NULL"

```suggestion
  bool is_progress = strequal(msg_ext_kind, MSG_KIND_PROGRESS);
```